protected/models/CurationLog.php

```diff
@@ -81,29 +84,48 @@ public function attributeLabels()
             'last_modified_by' => 'Last Modified By',
         );
     }
-    
-    public static function createlog($status,$id) {
-       
-        $curationlog = new CurationLog;
+
+    /**
+     * Factory method to make a new instance and factor out the common code
+     *
+     * @param int $id
+     * @param string $creator
+     * @return CurationLog
+     */
+    public static function makeNewInstanceForCurationLogBy(int $id, string $creator): CurationLog
+    {
+        $curationlog = new CurationLog();
         $curationlog->creation_date = date("Y-m-d");
         $curationlog->last_modified_date = null;
         $curationlog->dataset_id = $id;
-        $curationlog->created_by = "System";
+        $curationlog->created_by = $creator;
+        return $curationlog;
+    }
+
+    public static function createlog($status,$id) {
+
+        $curationlog = self::makeNewInstanceForCurationLogBy($id, "System");
         $curationlog->action = "Status changed to ".$status;
-        if (!$curationlog->save())
-            return false;
+        return $curationlog->save();
     }
-    
+
     public static function createlog_assign_curator($id,$creator,$username) {
 
-        $curationlog = new CurationLog;
-        $curationlog->creation_date = date("Y-m-d");
-        $curationlog->last_modified_date = null;
-        $curationlog->dataset_id = $id;
-        $curationlog->created_by = $creator;
+        $curationlog = self::makeNewInstanceForCurationLogBy($id, $creator);
         $curationlog->action = "Curator Assigned"." $username";
-        if (!$curationlog->save())
-            return false;
+        return $curationlog->save();
+    }
+
+    /**
+     * Retrieves attributes and store them in curation_log table when triggered
+     * @param $id
+     * @return bool
+     */
+    public static function createCurationLogEntry(int $id): bool
+    {
+        $curationlog = self::makeNewInstanceForCurationLogBy($id, "System");
+        $curationlog->action = "Status changed to stuff";
+        return $curationlog->save();
     }
```

Comment:
Is this `createCurationLogEntry(int $id)` function required? The `$curationlog->action = "Status changed to stuff";` line suggests it is not required.

Reply:
`createCurationLogEntry(int $id)` is needed for `DeleteFileAttributeAction`, which is responsible to delete file attribute.

Reply:
The function has been updated to `createCurationLogEntry(int $id, string $fileName)`, then it could save delete message in the `action column` in `curation_log` table.